src/parsing/literalParser.ts

```diff
@@ -200,23 +203,17 @@ export const dirCommentContents: Parser<string> = map(
 );
 
 export const dirCommentConstructor: Parser<IAST> = map(
-	delimited(tokens.DIR_COMMENT_OPEN, dirCommentContents, tokens.DIR_COMMENT_CLOSE),
+	delimited(tokens.DIR_COMMENT_OPEN, cut(dirCommentContents), tokens.DIR_COMMENT_CLOSE),
 	(x) => ['computedCommentConstructor', ['argExpr', ['stringConstantExpr', ['value', x]]]]
 );
 
-const piTarget: Parser<string> = preceded(
-	peek(
-		not(
-			then3(
-				or([tokens.X_UPPER, tokens.X_LOWER]),
-				or([tokens.M_UPPER, tokens.M_LOWER]),
-				or([tokens.L_UPPER, tokens.L_LOWER]),
-				(_a, _b, _c) => []
-			),
-			[]
-		)
-	),
-	name
+// Note: we deviate from the spec here. Processing instruction targets must _always_ be a NCName
```

Comment:
It's a bit unfortunate that the grammar only links to the XML spec for this production, but the note in the spec linking to https://www.w3.org/TR/2017/REC-xquery-31-20170321/#parse-note-xml-version seems to suggest that the "Namespaces in XML" spec does apply here. I don't think this is necessarily a deviation from the spec...